lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -236,10 +250,10 @@ def nacls_changed(nacl, client, module):
     nacl_id = nacl['NetworkAcls'][0]['NetworkAclId']
     nacl = describe_network_acl(client, module)
     entries = nacl['NetworkAcls'][0]['Entries']
-    tmp_egress = [entry for entry in entries if entry['Egress'] is True and DEFAULT_EGRESS != entry]
+    tmp_egress = [entry for entry in entries if entry['Egress'] is True and not match_default_rules(entry, True)]
```

Comment:
How about using a simple integer comparison for the RuleNumber, similar to what is done in https://github.com/ansible/ansible/pull/30581/files?

```suggestion
    tmp_egress = [entry for entry in entries if entry['Egress'] is True and entry['RuleNumber'] < 32767
```
I am not sure what was the rationale to add a DEFAULT_EGRESS rule.